data/get_hymns.py

```diff
@@ -25,10 +26,36 @@
 url = f"{base_url}?{query_string}"
 
 hymns = pd.read_csv(url)
-hymns = hymns[['displayTitle', 'authors']]
+
+hymns = hymns[['displayTitle', 'authors', 'textAuthNumber']]
+
 hymns = hymns.rename(columns={'displayTitle': 'title'})
 
 hymns.replace({np.nan: None}, inplace=True)
-        
-# export to json
-hymns.to_json("hymns.json", orient="records", indent=2)
\ No newline at end of file
+
+# Retrieve texts
+texts_url = "https://hymnary.org/files/hymnary/other/texts.csv"
+texts = pd.read_csv(texts_url)
+
+merged = pd.merge(hymns, texts[['textAuthNumber', 'yearsWrote']], on='textAuthNumber', how='left')
+
+# File write and appending to JSON file
+results = []
+for _, row in merged.iterrows():
+    title = str(row.get("title", "")).strip()
+    authors = str(row.get("authors", "")).strip()
+    pub_date = (
+        str(row.get("yearsWrote", "")).strip()
+        if pd.notna(row.get("yearsWrote"))
+        else None
+    )
+
+    results.append({
+        "title": title,
+        "authors": authors,
+        "publicationDate": pub_date
+    })
+
```

Comment:
Something like this to clean, filter, and rename our columns right before writing to JSON seems sensible to me:

```suggestion

hymns.replace({np.nan: None}, inplace=True)
hymns = hymns.rename(columns={
    'displayTitle': 'title',
    'yearsWrote': 'publicationDate',
})
hymns = hymns[['title', 'authors', 'publicationDate']]
```